Sources/PackageGraph/PackageGraph.swift

```diff
@@ -80,9 +80,20 @@ public struct PackageGraph {
     public let requiredDependencies: Set<PackageReference>
 
     /// Returns true if a given target is present in root packages.
-    public func isInRootPackages(_ target: ResolvedTarget) -> Bool {
+    public func isInRootPackages(_ target: ResolvedTarget, buildEnvironment: BuildEnvironment) -> Bool {
```

Comment:
I would maybe separate this out to a new function "includedInBuildEnviroment" or so, and then filter in the call site like we do for test targets. this way this function stays more true to its original intent

Reply:
I think I'll just rename the existing function since there aren't any clients that wouldn't want to pass in a build env today. Also potentially want to address the FIXME about caching since the computation has gotten more expensive?

Reply:
Renamed the method now.